src/components/Attributes/AttributeRow.tsx

```diff
@@ -0,0 +1,294 @@
+import { TextField } from "@material-ui/core";
+import makeStyles from "@material-ui/core/styles/makeStyles";
+import { AttributeInput } from "@saleor/components/Attributes/Attributes";
+import BasicAttributeRow from "@saleor/components/Attributes/BasicAttributeRow";
+import ExtendedAttributeRow from "@saleor/components/Attributes/ExtendedAttributeRow";
+import FileUploadField, {
+  FileChoiceType
+} from "@saleor/components/FileUploadField";
+import MultiAutocompleteSelectField, {
+  MultiAutocompleteChoiceType
+} from "@saleor/components/MultiAutocompleteSelectField";
+import SingleAutocompleteSelectField, {
+  SingleAutocompleteChoiceType
+} from "@saleor/components/SingleAutocompleteSelectField";
+import SortableChipsField, {
+  SortableChipsFieldValueType
+} from "@saleor/components/SortableChipsField";
+import { AttributeValueFragment } from "@saleor/fragments/types/AttributeValueFragment";
+import { PageErrorWithAttributesFragment } from "@saleor/fragments/types/PageErrorWithAttributesFragment";
+import { ProductErrorWithAttributesFragment } from "@saleor/fragments/types/ProductErrorWithAttributesFragment";
+import { FormsetChange } from "@saleor/hooks/useFormset";
+import { ReorderEvent } from "@saleor/types";
+import { AttributeInputTypeEnum } from "@saleor/types/globalTypes";
+import { getProductErrorMessage } from "@saleor/utils/errors";
+import getPageErrorMessage from "@saleor/utils/errors/page";
+import React from "react";
+import { defineMessages, IntlShape, useIntl } from "react-intl";
+
+function getSingleChoices(
```

Comment:
All of these functions should be moved to utils

Reply:
I thought u'll yell if I mess too much with ur code 👀